panther_hardware_interfaces/test/panther_system/gpio/test_gpio_controller.cpp

```diff
@@ -57,118 +64,129 @@ class TestGPIOController : public ::testing::Test
 protected:
   float GetRobotVersion();
 
+  std::shared_ptr<MockGPIODriver> gpio_driver_;
   std::unique_ptr<GPIOControllerWrapper> gpio_controller_wrapper_;
   static constexpr int watchdog_edges_per_100ms_ = 10;
 };
 
 TestGPIOController::TestGPIOController()
 {
-  if (GetRobotVersion() < 1.2 - std::numeric_limits<float>::epsilon()) {
-    throw std::runtime_error("Tests for this robot versions are not implemented");
-  }
+  gpio_driver_ = std::make_shared<MockGPIODriver>();
 
-  gpio_controller_wrapper_ = std::make_unique<GPIOControllerWrapper>();
+  // Mock methods called during the initialization process
+  ON_CALL(*gpio_driver_, SetPinValue(GPIOPin::VMOT_ON, true)).WillByDefault(testing::Return(true));
+  ON_CALL(*gpio_driver_, SetPinValue(GPIOPin::DRIVER_EN, true))
+    .WillByDefault(testing::Return(true));
+  ON_CALL(*gpio_driver_, IsPinAvailable(GPIOPin::WATCHDOG)).WillByDefault(testing::Return(true));
```

Comment:
Why watchdog state is not set and not checked in TestMotorsInitSuccess?

Reply:
Since this pin is not set during the initialization